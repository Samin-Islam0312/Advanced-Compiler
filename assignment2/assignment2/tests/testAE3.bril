# ARGS: ae
@main {
.entry:
  a: int = const 1;
  b: int = const 2;
  c: bool = gt a b;
  br c .then .else;
.then:
  t1: int = add a b;     # computes expr
  jmp .join;
.else:
  t2: int = add b a;     # same canonical expr
  jmp .join;
.join:
  u: int = add a b;      # available at join entry (on all paths)
  print u;
}

