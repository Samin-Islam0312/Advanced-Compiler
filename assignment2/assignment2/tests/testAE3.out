entry:
  in:  âˆ…
  out: gt(a, b)
then:
  in:  gt(a, b)
  out: add(a, b), gt(a, b)
else:
  in:  gt(a, b)
  out: add(a, b), gt(a, b)
join:
  in:  add(a, b), gt(a, b)
  out: add(a, b), gt(a, b)
